### PR TITLE
feat(iroh-cli): add file logging by default for start commands

### PR DESCRIPTION
**Pull request by divagant-martian**
## Description

Adds file logging by default to commands that start a node.

Parameters are:
- the "RUST_LOG" directive, in the form of and env var `IROH_FILE_RUST_LOG` or in the configuration file.
- rotation: how often a new file is created. Either daily, hourly or never.
- max files: Maximum number of files to keep

File logging can be disabled by either setting the max_files to 0 in the configuration file or by setting the rust log statement to off, in either the env var or the config file.

(example config file)
```toml
[file_logs]
rust_log = "iroh=trace"
rotation = "daily"
max_files = 1
```

The files created live the `IROH_DATA_DIR/logs` dir and are named `iroh-2024-03-02.log` (for a daily rotation, or with the hour, for an hourly rotation)

By default, we log everything in debug, rotating hourly, keeping at most 4 files (so total 4 hours of logs). This is short default since debug logs, even in an idle node, are a lot

## Notes & open questions

n/a

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

File: iroh-cli/src/commands.rs
Comment (flub) [inline iroh-cli/src/commands.rs:1]: Should `Cli::log_fd` on line 43 also be removed?

File: iroh-cli/Cargo.toml [hunk 1]
```diff
@@ -57,6 +57,8 @@ tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 tokio = { version = "1.36.0", features = ["full"] }
 tempfile = "3.10.1"
 flume = "0.11.0"
+tracing-appender = "0.2.3"
+serde_with = "3.7.0"
```
[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/Cargo.toml:61]: cargo add added them like this. I suspect it's due to the import just above these (`flume`) which is out of order

[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/Cargo.toml:61]: Actually, it seems to be `flume`, `postcard`, `serde`, `tempfile`, `tokio`, `tokio-util`, `tracing`, `tokio`, `tokio-util` are all out of order.

[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/Cargo.toml]: sorted in 5047341dd2a706f43301fc25269f9a469dd29b05

File: iroh-cli/src/commands.rs
Comment (divagant-martian) [reply iroh-cli/src/commands.rs:1]: yes, thanks. Rmoved in 1571dd4

File: iroh-cli/src/logging.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        // validate the RUST_LOG statement
+        let _valid_env = tracing_subscriber::EnvFilter::from_str(s)?;
+        Ok(EnvFilter(s.into()))
+    }
+}
+
+impl Default for EnvFilter {
+    fn default() -> Self {
+        // rustyline is annoying
+        Self("rustyline=warn,debug".into())
```
[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/src/logging.rs:132]: done in b71d893

File: iroh-cli/src/logging.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+impl EnvFilter {
+    pub(crate) fn layer(&self) -> tracing_subscriber::EnvFilter {
+        tracing_subscriber::EnvFilter::from_str(&self.0).expect("validated RUST_LOG statement")
+    }
+}
+
+/// Hoe often should a new file be created for file logs.
+/// Akin to [`tracing_appender::rolling::Rotation`].
```
[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/src/logging.rs:144]: done in b71d893

File: iroh-cli/src/logging.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (divagant-martian) [reply iroh-cli/src/logging.rs]: done in 5a7d63d

**Review by flub — APPROVED** [review]
nice!

**Pull request merged**